### ISSUE TITLE
Fyne compiled application not working on another Mac M1

### ISSUE THREAD

**Issue by mezimm** [fyne-io/fyne#3747]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When I use "fyne package", the application runs without issue on my Mac. But when I send the application to my fellow developer, he gets this error message when he tries to run it (and the CRC32 checks out):

<img width="133" alt="error" src="https://user-images.githubusercontent.com/63455166/226143616-6e6db8b7-161d-4641-bc37-4eb18b42b603.png">

The reverse is also true. When he compiles it using "fyne package" on his system, sends it to me, and I try to run it, the same thing happens.

We are both on Mac M1 Pro with Ventura 13.2.1 installed. We did not notice this issue a week or two ago. I believe it just started happening with the latest Mac update. Also, when we build just the executable file using "go build ." we no longer have this issue. But obviously, that is not ideal since we lose the icon and other supplied files in the package.

I noticed these two threads, and investigated them, but they do not seem to share this issue.
https://github.com/fyne-io/fyne/issues/3743
https://github.com/fyne-io/fyne/issues/3639

### How to reproduce

1. Build your application with "fyne package" on a Mac M1
2. Share your application to another person on another Mac M1
3. Try to run the application

### Example code

fyne package

### Fyne version

2.3.1

### Go compiler version

1.20

### Operating system and version

MacOS Ventura 13.2.1 on M1

**Comment by andydotxyz** [comment]
macOS has marked it as being in quarantine when downloaded. You can remove this flag and the app will work. Or it could be signed (if you pay Apple money!)

https://superuser.com/questions/526920/how-to-remove-quarantine-from-file-permissions-in-os-x

**Comment by andydotxyz** [comment]
TL;DR it isn't damaged, Apple is pushing you to pay for distribution. Using xattr manages to avoid it, but per-install :(

**Comment by mezimm** [comment]
Thank you all. I will proceed with signing and see if that solves it.
